posts/deploying-a-jhipster-app-to-aws-using-elastic-beanstalk.md

```diff
@@ -0,0 +1,101 @@
+---
+authors:
+- Theo Lebrun
+tags:
+- JHipster
+- AWS
+- Elastic Beanstalk
+date: 2019-11-04T14:50:55.000Z
+title: "Deploying a JHipster app to AWS using Elastic Beanstalk"
+image: https://raw.githubusercontent.com/falydoor/blog-usa/jhipster-on-aws-with-beanstalk/images/2019/11/jhipster-beanstalk.png
+---
+
+[JHipster](https://www.jhipster.tech/) is a great development platform to help you bootstrap a modern Web app in less than 5 minutes. One small detail is that everything runs on your local laptop which can be an issue if you want to showcase your app to external people. Cloud providers like [AWS](https://aws.amazon.com/), [Azure](https://azure.microsoft.com) or [GCP](https://cloud.google.com/) are the current leaders and they provide services to help you deploy your JHipster app to the Cloud.
+
+In this blog, I will focus on AWS and especially on Elastic Beanstalk which is an orchestration service. Elastic Beanstalk helps you create a production-ready infrastructure that has the architecture below:
+
+![](https://raw.githubusercontent.com/falydoor/blog-usa/jhipster-on-aws-with-beanstalk/images/2019/11/diagram-eb.png)
+
+# App generation with JHipster
+
+Currently the Elastic Beanstalk sub-generator only supports SQL databases (Oracle and Microsoft SQL Server excluded). Our JHipster app will be a basic monolith that uses React and PostgreSQL as a database.
```

Comment:
```suggestion
Fortunately, the newest versions of JHipster feature an Elastic Beanstalk sub-generator which will do most of the heavy-lifting for us. It does currently only support SQL databases (Oracle and Microsoft SQL Server excluded) so our JHipster app will be a basic monolith that uses React and PostgreSQL as a database. To avoid answering all the generator's questions manually, feel free to utilize the following `.yo-rc.json` with our proposed setup:
```